### PR TITLE
bpo-29386: Cleanup select_epoll_poll_impl()

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://bugs.python.org/issue29386

### CODE REVIEW

File: Modules/selectmodule.c [hunk 1]
```diff
@@ -1498,17 +1498,12 @@ select_epoll_poll_impl(pyEpoll_Object *self, PyObject *timeout_obj,
...
     if (self->epfd < 0)
         return pyepoll_err_closed();
 
-    if (timeout_obj == Py_None) {
-        timeout = -1;
-        ms = -1;
-        deadline = 0;   /* initialize to prevent gcc warning */
-    }
-    else {
+    if (timeout_obj != NULL && timeout_obj != Py_None) {
```
Comment (serhiy-storchaka) [inline Modules/selectmodule.c]: It is never NULL.

File: Modules/selectmodule.c [hunk 2]
```diff
@@ -1526,7 +1521,19 @@ select_epoll_poll_impl(pyEpoll_Object *self, PyObject *timeout_obj,
...
+        }
+    }
+
+    /* Kernel doesn't make any difference for negative values, but -1 is
+       documented as a way to block indefinitely in the epoll_wait()
+       documentation, so set ms to -1 if the value is negative.
+
+       Note that we didn't use INFTIM here since it's non-standard and
+       doesn't available under Linux. */
+    if (ms < 0) {
```
Comment (serhiy-storchaka) [inline Modules/selectmodule.c]: It is better to move this check above, closer to other check for `ms`.

File: Modules/selectmodule.c (hunk 1 above)
Comment (berkerpeksag) [reply Modules/selectmodule.c]: Ah, I was looking at the old commit and I missed the Argument Clinic conversion.

File: Modules/selectmodule.c (hunk 2 above)
Comment (berkerpeksag) [reply Modules/selectmodule.c]: Moved and tweaked the comment a bit.

**Review by serhiy-storchaka — APPROVED** [review]

File: Modules/selectmodule.c [hunk 3]
```diff
@@ -1525,8 +1520,20 @@ select_epoll_poll_impl(pyEpoll_Object *self, PyObject *timeout_obj,
             PyErr_SetString(PyExc_OverflowError, "timeout is too large");
             return NULL;
         }
+        /* epoll_wait(2) treats all arbitrary negative numbers the same
+           for the timeout argument, but -1 is the documented way to block
+           indefinitely in the epoll_wait(2) documentation, so we set ms
+           to -1 if the value of ms is a negative number.
+
+           Note that we didn't use INFTIM here since it's non-standard and
+           doesn't available under Linux. */
```
Comment (ZackerySpytz) [inline Modules/selectmodule.c]: It should be "isn't", not "doesn't".

Comment (berkerpeksag) [reply Modules/selectmodule.c]: Good catch!

**Pull request merged**